src/base/bittorrent/torrentimpl.cpp

```diff
@@ -236,6 +236,14 @@ namespace
         status.pieces = params.have_pieces;
         status.verified_pieces = params.verified_pieces;
     }
+
+    template <typename Vector>
+    Vector resized(const Vector &inVector, const typename Vector::size_type size, const typename Vector::value_type &defaultValue)
+    {
+        Vector outVector = inVector;
```

Comment:
I would do the following to avoid redundant copy from inVector:
```c++
Vector outVector(size, defaultValue);
std::copy_n(inVector.begin(), size, std::back_inserter(outVector));
return outVector;
```

Reply:
> I would do the following to avoid redundant elements from inVector

Having "redundant elements from inVector" is a less likely scenario.
But maybe I'll improve it in some other way in master.